### ISSUE TITLE
Add optional version param to jsDelivr badges

### ISSUE THREAD

**Issue by chris48s** [badges/shields#2981]
:clipboard: **Description**

It would be nice if we could call for example `/hd/jquery/jquery/3.3.1.svg`

**Comment by Prouser123** [comment]
How would we go about implementing this?
Asking because of an in-progress refactor, but I'm also happy to take this up.

**Comment by chris48s** [comment]
An optional param can be added by using a `?` e.g:

https://github.com/badges/shields/blob/951aa0f9ead34637ac5843bb1316d434d1aa0a58/services/jenkins/jenkins-plugin-installs.service.js#L97-L102

so you could modify the pattern in

https://github.com/badges/shields/blob/951aa0f9ead34637ac5843bb1316d434d1aa0a58/services/jsdelivr/jsdelivr-hits-github.service.js#L6-L11

to 

```js
static get route() {
  return {
    base: 'jsdelivr/gh',
    pattern: ':period(hd|hw|hm|hy)/:user/:repo/:version?',
  }
}
```

If I call

https://data.jsdelivr.com/v1/package/gh/jquery/jquery/stats/date/week

for example, I could write some code like

```js
const data = await this.fetch(/*params*/)
const version = '3.3.1'
let downloads = 0
for (date in data.dates) {
  if (version in data.dates[date].versions) {
    downloads += data.dates[date].versions[version]
  }
}
```

to iterate through each date and sum the downloads for only a specific version.

**Comment by MartinKolarik** [comment]
Or just ask for that version 😛 https://data.jsdelivr.com/v1/package/gh/jquery/jquery@3.3.1/stats/date/week

**Comment by chris48s** [comment]
..or that :smile:

**Comment by Prouser123** [comment]
So currently we can call `jsdelivr/npm/hw/jquery@3.3.1.svg` to get only that version, but would it be worth either adding some documentation to show this or adding an optional version parameter to make it easier?

**Comment by chris48s** [comment]
Yes - this is a good point - you could just add an example to the examples array to show calling with the `package@version` syntax. I hadn't clocked that would work as the code stands.

One other thing we usually do with this style of badge is modify the label if version is specified. e.g:

![](https://img.shields.io/crates/d/rustc-serialize.svg) - https://img.shields.io/crates/d/rustc-serialize.svg
![](https://img.shields.io/crates/d/rustc-serialize/0.3.24.svg) - https://img.shields.io/crates/d/rustc-serialize/0.3.24.svg

![](https://img.shields.io/jenkins/plugin/i/view-job-filters.svg) - https://img.shields.io/jenkins/plugin/i/view-job-filters.svg
![](https://img.shields.io/jenkins/plugin/i/view-job-filters/1.26.svg) - https://img.shields.io/jenkins/plugin/i/view-job-filters/1.26.svg

**Comment by paulmelnikow** [comment]
Maybe it’s better to support it with a slash as usual, and use a stricter regex on the version that rejects it as part of the package name. That’s a bit friendlier in the UI which is the main reason; it also requires less custom logic to detect the version and render the label.
